tests/api/test_topic_entity_tag.py

```diff
@@ -0,0 +1,218 @@
+from .fixtures import auth_headers, db # noqa
+from .test_reference import test_reference # noqa
+
+TestTETData = namedtuple('TestTETData', ['response', 'new_tet_id', 'related_ref_curie'])
+
+
+@pytest.fixture
+def test_topic_entity_tag(db, auth_headers, test_reference): # noqa
+    print("***** Adding a test workflow tag *****")
+    with TestClient(app) as client:
+        new_tet = {
+            "reference_curie": test_reference.json(),
+            "topic": "Topic1",
+            "entity_type": "Gene",
+            "alliance_entity": "Bob_gene_name",
+            "taxon": "NCBITaxon:1234",
+            "note": "Some Note",
+            "props": [{"qualifier": "Quali1"},
+                      {"qualifier": "Quali2"}]
+        }
+        response = client.post(url="/topic_entity_tag/", json=new_tet, headers=auth_headers)
+        yield TestTETData(response, response.json(), test_reference.json())
+
+
+class TestTopicEntityTag:
+
+    def test_good_create_with_props(self, db, test_topic_entity_tag): # noqa
+        with TestClient(app) as client:
+            assert test_topic_entity_tag.response.status_code == status.HTTP_201_CREATED
+
+            tet_obj = db.query(TopicEntityTagModel).filter(
+                TopicEntityTagModel.topic_entity_tag_id == test_topic_entity_tag.new_tet_id).first()
+
+            # assert tet_obj.reference_id == refs[0].reference_id
+            assert tet_obj.topic == "Topic1"
+            assert tet_obj.entity_type == "Gene"
+            assert tet_obj.alliance_entity == "Bob_gene_name"
+            assert tet_obj.taxon == "NCBITaxon:1234"
+            assert tet_obj.note == "Some Note"
+
+            props = db.query(TopicEntityTagPropModel).filter(
+                TopicEntityTagPropModel.topic_entity_tag_id == test_topic_entity_tag.new_tet_id).all()
+
+            count = 0
+            for prop in props:
+                if prop.qualifier in ["Quali1", "Quali2"]:
+                    count += 1
+                else:
+                    assert "Diff qualifier" == prop.qualifier
+            assert count == 2
+
+            response = client.get(url=f"/topic_entity_tag/{test_topic_entity_tag.new_tet_id}")
+            res = response.json()
+            assert res["topic"] == "Topic1"
+            assert res["props"][0]["qualifier"] == "Quali1"
+            assert res["props"][1]["qualifier"] == "Quali2"
+
+    def test_create_bad(self, test_reference, auth_headers): # noqa
+        with TestClient(app) as client:
+            xml = {
+                "reference_curie": test_reference.json(),
+                "topic": "Topic1",
+                "entity_type": "Gene",
+                "taxon": "NCBITaxon:1234"
+            }
+            # No Entities
+            response = client.post(url="/topic_entity_tag/", json=xml, headers=auth_headers)
+            assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+
+            # More than one Entity
+            xml2 = copy.deepcopy(xml)
```

Comment:
xml is already 422.  Might it be better to start with a working xml, test that it works, then deepcopy that working xml and modify it to break in different ways to test that it's breaking because of the changes, instead of the first one broke, so all copies of it break regardless of different reasons for breaking ?

Reply:
I copied this from the old unit tests, but what you are saying makes sense. I'll rework this test following your suggestion

Reply:
We looked at this together and realized that the tests are ok, even though they are a bit confusing. We decided to keep the code as is